components/admin_console/admin_settings.tsx

```diff
@@ -0,0 +1,258 @@
+        this.setState({
+            saving: true,
+            serverError: null,
+        });
+
+        // clone config so that we aren't modifying data in the stores
+        let config = JSON.parse(JSON.stringify(this.props.config));
+        config = this.getConfigFromState(config);
+
+        if (this.props.updateConfig) {
+            const {data, error} = await this.props.updateConfig(config);
+
+            if (data) {
+                this.setState(this.getStateFromConfig(data));
+
+                this.setState({
+                    saveNeeded: false,
+                    saving: false,
+                });
+
+                if (this.props.setNavigationBlocked) {
+                    this.props.setNavigationBlocked(false);
+                }
+
+                if (callback) {
+                    callback();
+                }
+
+                if (this.handleSaved) {
+                    this.handleSaved(config);
+                }
+            } else if (error) {
+                this.setState({
+                    saving: false,
+                    serverError: error.message,
+                    serverErrorId: error.server_error_id,
+                });
+
+                if (callback) {
+                    callback();
+                }
+
+                if (this.handleSaved) {
+                    this.handleSaved(config);
+                }
+            }
+        }
+    };
+
+    private parseInt = (str: string, defaultValue?: number) => {
+        const n = parseInt(str, 10);
+
+        if (isNaN(n)) {
+            if (defaultValue) {
+                return defaultValue;
+            }
+            return 0;
+        }
+
+        return n;
+    };
+
+    private parseIntNonNegative = (str: string, defaultValue?: number) => {
+        const n = parseInt(str, 10);
+
+        if (isNaN(n) || n < 0) {
+            if (defaultValue) {
+                return defaultValue;
+            }
+            return 0;
+        }
+
+        return n;
+    };
+
+    private parseIntNonZero = (str: string, defaultValue?: number, minimumValue = 1) => {
+        const n = parseInt(str, 10);
+
+        if (isNaN(n) || n < minimumValue) {
+            if (defaultValue) {
+                return defaultValue;
+            }
+            return 1;
+        }
+
+        return n;
+    };
+
+    private getConfigValue(config: object, path: string) {
+        const pathParts = path.split('.');
+
+        return pathParts.reduce((obj: object|null, pathPart) => {
+            if (!obj) {
+                return null;
+            }
+            return obj[(pathPart as keyof object)];
+        }, config);
+    }
+
+    private setConfigValue(config: object, path: string, value: any) {
+        function setValue(obj: object, pathParts: string[]) {
+            const part = pathParts[0] as keyof object;
+
+            if (pathParts.length === 1) {
+                Object.assign(obj, {[part]: value});
```

Comment:
Why using this approach?

Reply:
I do not know the details on why it works like that, but if you use the original notation:

`obj[part] = value;`

Typescript finds the following type error:

`Type 'any' is not assignable to type 'never'.`

The only way I found around this is this one. If you know any other cleaner way to do this, or the reason why it behaves like that, it would be great to know.

Thanks!